### PR TITLE
Feature pagination and post functionalities

### PR DESCRIPTION
**Pull request by tejasvikolhe01**
Hi Team,

1. Added pagination
2. Sorted post by created date.

Thank you

### CODE REVIEW

**Review by MattX23 — CHANGES_REQUESTED** [review]
The pagination looks to be working well.

One thing though - I would clear the input after the comment has been submitted:

Screenshot 2024-07-09 at 14 48 26

File: resources/js/Pages/Dashboard.vue [hunk 1]
```diff
@@ -74,9 +75,17 @@ import { ref } from 'vue';
                                 </PrimaryButton>
                             </div>
                         </div><!-- post-item end -->
-                        <PrimaryButton v-if="this.posts.length>3 && this.postLength < this.posts.length" @click="loadMore" class="offset-lg-5 col-md-3">
-                            Load More
-                        </PrimaryButton>
+                        <div>
+                            <a  href="javascript:void(0)" @click="loadMorePosts(link.url)">{{ index }}</a>
```
Comment (alex-bowers) [inline resources/js/Pages/Dashboard.vue:79]: Why do you need `javascript:void(0)`?

File: resources/js/Pages/Dashboard.vue [hunk 2]
```diff
@@ -74,9 +75,17 @@ import { ref } from 'vue';
...
+                        </div>
+                        <!-- pagination -->
+                        <nav aria-label="Page">
+                            <ul class="pagination float-end">
+                                <li v-for="(link,index) in loadMoreLinks"  class="page-item">
+                                    <a v-bind:class="{ active: link.active }" class="page-link" @click="loadMorePosts(link.url)" v-if="index>0 && index<loadMoreLinks.length-1" href="javascript:void(0)">{{ index }}</a>
```
Comment (alex-bowers) [inline resources/js/Pages/Dashboard.vue:85]: You don't need to type `v-bind` in this instance. Just typing `:class` works the same way.

File: resources/js/Pages/Dashboard.vue [hunk 3]
```diff
@@ -90,38 +99,56 @@ import { ref } from 'vue';
...
                 this.post_content='';
             },
             // fetch all posts
             getAllPosts() {
                 axios
                 .get('/posts')
-                .then(response => (this.posts = response.data))
+                .then(response => {
+                    this.posts = response.data.data;
+                    this.loadMoreLinks = response.data.links;
```
Comment (alex-bowers) [inline resources/js/Pages/Dashboard.vue:134]: Can you destruct `response.data`?

**Review by MattX23 — APPROVED** [review]

**Review by alex-bowers — APPROVED** [review]

**Pull request merged**